File: linked-list/return-k-element/solution_v1.go
package return_k_element

type Node struct {
	Next  *Node
	value int
}

func ReturnKElementV1(node *Node, k int) int {
	total := 1

	for {
		if node == nil {
			return -1
		}

		if total == k {
			return node.value
		}

		total += 1
		node = node.Next
	}

	return -1
}

File: linked-list/loop-detection-java/src/main/java/SolutionV1.java
import java.util.HashSet;
import java.util.Set;

public class SolutionV1 {

    public Character loopDetection(Node node) {
        Set<Character> uniques = new HashSet<>();

        return helper(node, uniques);
    }

    private Character helper(Node node, Set<Character> uniques) {
        if (node.next == null){
            return null;
        }

        if (uniques.contains(node.data)) {
            return node.data;
        }

        uniques.add(node.data);
        return helper(node.next, uniques);
    }
}
